Form/FieldTypeHandler/IntegerHandler.php

```diff
@@ -23,8 +41,14 @@ protected function buildFieldForm(
     ) {
         $options = $this->getDefaultFieldOptions($fieldDefinition, $languageCode, $content);
 
-        if (!empty($fieldDefinition->defaultValue)) {
-            $options['data'] = (int)$fieldDefinition->defaultValue->value;
+        if ($fieldDefinition->defaultValue instanceof IntegerValue\Value) {
+            if ($content instanceof Content) {
+                if ($this->fieldHelper->isFieldEmpty($content, $fieldDefinition->identifier, $languageCode)) {
+                    $options['data'] = (int)$fieldDefinition->defaultValue->value;
```

Comment:
This is the case for updating, right (`$content` being set)? Why do you set the default value if the field is empty in this case? AFAICS it should never be set for update.